### PR TITLE
Remotes/emqx/release 3.2

### PR DESCRIPTION
**Pull request by Anakin-Hao**

### CODE REVIEW

File: src/emqx_auth_http.erl [hunk 1]
```diff
@@ -36,8 +40,10 @@ register_metrics() ->
 check(Credentials, #{auth_req := AuthReq,
                      super_req := SuperReq,
                      http_opts := HttpOpts,
-                     retry_opts := RetryOpts}) ->
-    case authenticate(AuthReq, Credentials, HttpOpts, RetryOpts) of
+                     retry_opts := RetryOpts,
+                     params := Params}) ->
```
Comment (PeiyiChi) [inline src/emqx_auth_http.erl]: Shouldn't add a parameter on callback function.

File: src/emqx_auth_http.erl [hunk 2]
```diff
@@ -55,6 +61,9 @@ check(Credentials, #{auth_req := AuthReq,
             {stop, Credentials#{auth_result => Error, anonymous => false}}
     end.
 
+check(#{username := Username}, Password, _Env) when ?UNDEFINED(Username); ?UNDEFINED(Password) ->
```
Comment (PeiyiChi) [inline src/emqx_auth_http.erl]: Please remove this block if the new code doesn't need it.

File: src/emqx_auth_http.erl [hunk 3]
```diff
@@ -20,13 +20,17 @@
 -include_lib("emqx/include/logger.hrl").
 
 -import(emqx_auth_http_cli,
-        [ request/5
+        [request/3
```
Comment (PeiyiChi) [inline src/emqx_auth_http.erl]: Please remove it if we are not using request/3

File: src/emqx_auth_http_cli.erl [hunk 4]
```diff
@@ -27,6 +31,23 @@ request(get, Url, Params, HttpOpts, RetryOpts) ->
...
+    {_, Username} = lists:keyfind("username", 1, Params),
+    Realm = binary_to_list(lists:nth(1, string:split(Username, "_"))),
+
+    Auth = "Bearer " ++ binary_to_list(Token),
+    LoginUrl = string:join(string:replace(Url, "{}", Realm), ""),
+    Req = {LoginUrl, [{"Authorization", Auth}, {"User-Agent", "albi_internal"}]},
+    logger:info("Reaching out to id-rest to verify token"),
+    reply(request_(get, Req, [{autoredirect, true}], [], 0)).
+
+request(get, Url, Params) ->
```
Comment (PeiyiChi) [inline src/emqx_auth_http_cli.erl]: Please remove this function since it has been replaced by request(get, Url, Params, HttpOpts, RetryOpts)

**Comment by Anakin-Hao** [comment]
> Please revert this request function to request(get, Url, Params, HttpOpts, RetryOpts)

Resolved.

File: src/emqx_auth_http.erl (hunk 1 above)
Comment (Anakin-Hao) [reply src/emqx_auth_http.erl]: Fixed.

**Pull request merged**